core/config/project_settings.h

```diff
@@ -36,6 +36,7 @@
 #include "core/templates/hash_map.h"
 #include "core/templates/local_vector.h"
 #include "core/templates/rb_set.h"
+#include "core/variant/typed_array.h"
```

Comment:
Use a forward-declare instead and move the include to cpp:
```C++
template <typename T>
class TypedArray;
```

Reply:
I don't think we can have forward-declared class template members (as the compiler has to know the class size here). Returning or passing them as arguments seems to work fine. I could either:

* Keep the include and code as-is, or
* Use the forward declaration and change `TypedArray<Dictionary> global_class_list;` member variable into normal `Array global_class_list;` and then create and return TypedArray<Dictionary> from the get_global_class_list() only when it's called.

Reply:
I think it's fine if the member is Array. Converting it is not expensive and the function is not called often.